doc/api/async_hooks.md

```diff
@@ -0,0 +1,532 @@
+# Async Hooks
+
+> Stability: 1 - Experimental
+
+The `async_hooks` module provides an API to register callbacks tracking the
+lifetime of asynchronous resources created inside a Node.js application.
+It can be accessed using:
+
+```js
+const async_hooks = require('async_hooks');
+```
+
+## Terminology
+
+An asynchronous resource represents an object with an associated callback.
+This callback may be called multiple times, for example, the `connection` event
+in `net.createServer`, or just a single time like in `fs.open`. A resource
+can also be closed before the callback is called. AsyncHooks does not
```

Comment:
There are zero references to _request_ and _handle_ in the of the documentation (except for the previous terminology section). If _request_ and _handle_ are to be included, it should be described exactly how those concepts are visible in _async_hooks_. 

see: https://github.com/nodejs/node/pull/12953#discussion_r117768833